extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampCommentsExtractor.java

```diff
@@ -39,18 +51,77 @@ public InfoItemsPage<CommentsInfoItem> getInitialPage()
 
-        final Elements writings = document.getElementsByClass("writing");
+        final JsonObject collectorsData = JsonUtils.toJsonObject(
+                document.getElementById("collectors-data").attr("data-blob"));
+        final JsonArray reviews = collectorsData.getArray("reviews");
 
-        for (final Element writing : writings) {
-            collector.commit(new BandcampCommentsInfoItemExtractor(writing, getUrl()));
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
         }
 
-        return new InfoItemsPage<>(collector, null);
+        if (!collectorsData.getBoolean("more_reviews_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String trackId = getTrackId();
+        final String token = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, token));
     }
 
     @Override
     public InfoItemsPage<CommentsInfoItem> getPage(final Page page)
             throws IOException, ExtractionException {
-        return null;
+
+        final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(getServiceId());
+
+        final String trackId = page.getUrl();
+        final String token = page.getId();
+        final JsonObject reviewsData = fetchReviewsData(trackId, token);
+        final JsonArray reviews = reviewsData.getArray("results");
+
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
+        }
+
+        if (!reviewsData.getBoolean("more_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String nextToken = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, nextToken));
+    }
+
+    private JsonObject fetchReviewsData(final String trackId, final String token)
+            throws ParsingException {
+        try {
+            return JsonUtils.toJsonObject(getDownloader().postWithContentTypeJson(
+                    REVIEWS_API_URL,
+                    Collections.emptyMap(),
+                    JsonWriter.string().object()
+                            .value("tralbum_type", "t")
+                            .value("tralbum_id", trackId)
+                            .value("token", token)
+                            .value("count", 7)
+                            .array("exclude_fan_ids").end()
+                            .end().done().getBytes(StandardCharsets.UTF_8)).responseBody());
+        } catch (final IOException | ReCaptchaException e) {
+            throw new ParsingException("Could not fetch reviews", e);
+        }
+    }
+
+    private String getNextPageToken(final JsonArray reviews) throws ParsingException {
+        return reviews.stream().map(JsonObject.class::cast)
```

Comment:
You should ensure the type of the object before casting to avoid `ClassCastException`s (even if it should not happen in this case):

```suggestion
        return reviews.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
```